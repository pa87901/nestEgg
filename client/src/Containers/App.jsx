import React from 'react';
import { Provider } from 'react-redux';
import { Router, Route, Switch } from 'react-router-dom';
import store from '../store/configureStore';
import history from '../store/history';
import Portfolio from './Portfolio';
import Clipboard from './Clipboard';

const FourOhFour = () => <h1>404</h1>
// console.log('store', store, 'history', history);

const App = () => (
  <Provider store={store} >
    <div>
      <h1>Nest Egg World</h1>
      <div className="blotter-container">
      <Router history={history}>
        <Switch>
          <Route exact path="/" component={Portfolio}/>
          <Route path="/transactions" component={Clipboard} />
          <Route component={FourOhFour}/>
        </Switch>
      </Router>
      </div>
    </div>
  </Provider>
);

export default App;