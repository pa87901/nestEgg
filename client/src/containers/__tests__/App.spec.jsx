import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  const shallowComponent = shallow(<App />);

  test('App snapshot has not changed', () => {
    expect(shallowComponent).toMatchSnapshot();
  });
});