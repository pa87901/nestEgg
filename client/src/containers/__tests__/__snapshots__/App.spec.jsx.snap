// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot has not changed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <h1>
          Nest Egg World
        </h1>
        <div
          className="blotter-container"
        >
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "m6ef0s",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <div>
              <Navigation />
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  path="/holdings"
                />
                <Route
                  component={[Function]}
                  path="/transactions"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </div>
          </Router>
        </div>
      </div>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Nest Egg World
          </h1>,
          <div
            className="blotter-container"
          >
            <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <div>
                <Navigation />
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    path="/holdings"
                  />
                  <Route
                    component={[Function]}
                    path="/transactions"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>
            </Router>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nest Egg World",
          },
          "ref": null,
          "rendered": "Nest Egg World",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <div>
                <Navigation />
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    path="/holdings"
                  />
                  <Route
                    component={[Function]}
                    path="/transactions"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>
            </Router>,
            "className": "blotter-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <Navigation />
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    path="/holdings"
                  />
                  <Route
                    component={[Function]}
                    path="/transactions"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>,
              "history": Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "m6ef0s",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Navigation />,
                  <Switch>
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />
                    <Route
                      component={[Function]}
                      path="/holdings"
                    />
                    <Route
                      component={[Function]}
                      path="/transactions"
                    />
                    <Route
                      component={[Function]}
                    />
                  </Switch>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                      />,
                      <Route
                        component={[Function]}
                        path="/holdings"
                      />,
                      <Route
                        component={[Function]}
                        path="/transactions"
                      />,
                      <Route
                        component={[Function]}
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "exact": true,
                        "path": "/",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "path": "/holdings",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "path": "/transactions",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <h1>
            Nest Egg World
          </h1>
          <div
            className="blotter-container"
          >
            <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <div>
                <Navigation />
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    path="/holdings"
                  />
                  <Route
                    component={[Function]}
                    path="/transactions"
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>
              </div>
            </Router>
          </div>
        </div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Nest Egg World
            </h1>,
            <div
              className="blotter-container"
            >
              <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "m6ef0s",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <div>
                  <Navigation />
                  <Switch>
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />
                    <Route
                      component={[Function]}
                      path="/holdings"
                    />
                    <Route
                      component={[Function]}
                      path="/transactions"
                    />
                    <Route
                      component={[Function]}
                    />
                  </Switch>
                </div>
              </Router>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Nest Egg World",
            },
            "ref": null,
            "rendered": "Nest Egg World",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Router
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "m6ef0s",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
              >
                <div>
                  <Navigation />
                  <Switch>
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />
                    <Route
                      component={[Function]}
                      path="/holdings"
                    />
                    <Route
                      component={[Function]}
                      path="/transactions"
                    />
                    <Route
                      component={[Function]}
                    />
                  </Switch>
                </div>
              </Router>,
              "className": "blotter-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div>
                  <Navigation />
                  <Switch>
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />
                    <Route
                      component={[Function]}
                      path="/holdings"
                    />
                    <Route
                      component={[Function]}
                      path="/transactions"
                    />
                    <Route
                      component={[Function]}
                    />
                  </Switch>
                </div>,
                "history": Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "m6ef0s",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "m6ef0s",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Navigation />,
                    <Switch>
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                      />
                      <Route
                        component={[Function]}
                        path="/holdings"
                      />
                      <Route
                        component={[Function]}
                        path="/transactions"
                      />
                      <Route
                        component={[Function]}
                      />
                    </Switch>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <Route
                          component={[Function]}
                          exact={true}
                          path="/"
                        />,
                        <Route
                          component={[Function]}
                          path="/holdings"
                        />,
                        <Route
                          component={[Function]}
                          path="/transactions"
                        />,
                        <Route
                          component={[Function]}
                        />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "component": [Function],
                          "exact": true,
                          "path": "/",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "component": [Function],
                          "path": "/holdings",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "component": [Function],
                          "path": "/transactions",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "component": [Function],
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
